mlflow/server/job/__init__.py

```diff
@@ -0,0 +1,181 @@
+) -> str:
+    """
+    Submit a job to the job queue. The job is executed at most once.
+    If the MLflow server crashes while the job is pending or running,
+    it is rescheduled on restart.
+
+    Note:
+        This is a server-side API and requires the MLflow server to configure
+        the backend store URI to a database URI.
+
+    Note:
+        This is a server-side API, and it requires MLflow server configures
+        backend store URI to database URI.
+
+    Args:
+        function: The job function, it must be a python module-level function,
+            and all params and return value must be JSON-serializable.
+            The function can raise `TransientError` in order to trigger
+            job retry, you can set `MLFLOW_SERVER_JOB_TRANSIENT_ERROR_MAX_RETRIES`
+            to configure maximum allowed retries for transient errors
+            and set `MLFLOW_SERVER_JOB_TRANSIENT_ERROR_RETRY_BASE_DELAY` to
+            configure base retry delay in seconds.
+        params: The params to be passed to the job function.
+        timeout: (optional) the job execution timeout, default None (no timeout)
+
+    Returns:
+        The job id. You can call `query_job` API by the job id to get
+        the job status and result.
+    """
+    from mlflow.environment_variables import MLFLOW_SERVER_ENABLE_JOB_EXECUTION
+    from mlflow.server.job.job_runner import huey_task_exec_job
+
+    if not MLFLOW_SERVER_ENABLE_JOB_EXECUTION.get():
+        raise MlflowException(
+            "Mlflow server job execution feature is not enabled, please set "
+            "environment variable 'MLFLOW_SERVER_ENABLE_JOB_EXECUTION' to 'true' to enable it."
+        )
+
+    if not (isinstance(function, FunctionType) and "." not in function.__qualname__):
+        raise MlflowException("The job function must be a python global function.")
+
+    job_store = _get_job_store()
+    serialized_params = json.dumps(params)
+    func_fullname = f"{function.__module__}.{function.__name__}"
+    job = job_store.create_job(func_fullname, serialized_params, timeout)
+
+    # enqueue job
+    huey_task_exec_job(job.job_id, function, params, timeout)
+
+    return job.job_id
+
+
+def query_job(job_id: str) -> tuple[JobStatus, Any]:
+    """
+    Query the job status and result by the job id.
+
+    Note:
+        This is a server-side API, and it requires MLflow server configures
+        backend store URI to database URI.
+
+    Args:
+        job_id: The job id.
+
+    Returns:
+        A tuple of (status, result)
+        status value is one of PENDING / RUNNING / DONE / FAILED.
+        If status is PENDING / RUNNING, result is None.
+        If status is DONE, result is the job function returned value.
+        If status is FAILED, result is the error message.
+    """
+    job_store = _get_job_store()
+    job = job_store.get_job(job_id)
+    status = job.status
+    result = job.result
```

Comment:
```suggestion
```

This seems redundant, increases LOC, and makes us wonder "what stats/result". Let's remove them.